src/Sulu/Bundle/HttpCacheBundle/Tests/Unit/CacheLifetime/CacheLifetimeEnhancerTest.php

```diff
@@ -0,0 +1,122 @@
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\ResponseHeaderBag;
+
+class CacheLifetimeEnhancerTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var CacheLifetimeEnhancer
+     */
+    private $cacheLifetimeEnhancer;
+
+    /**
+     * @var CacheLifetimeResolver
+     */
+    private $cacheLifetimeResolver;
+
+    /**
+     * @var PageBridge
+     */
+    private $page;
+
+    /**
+     * @var SnippetBridge
+     */
+    private $snippet;
+
+    /**
+     * @var Response
+     */
+    private $response;
+
+    /**
+     * @var ResponseHeaderBag
+     */
+    private $responseHeaderBag;
+
+    /**
+     * @var int
+     */
+    private $maxAge = 200;
+
+    /**
+     * @var int
+     */
+    private $sharedMaxAge = 300;
+
+    public function setUp()
+    {
+        $this->cacheLifetimeResolver = $this->prophesize(CacheLifetimeResolver::class);
+        $this->page = $this->prophesize(PageBridge::class);
+        $this->snippet = $this->prophesize(SnippetBridge::class);
+        $this->response = $this->prophesize(Response::class);
+        $this->responseHeaderBag = $this->prophesize(ResponseHeaderBag::class);
+        $this->response->headers = $this->responseHeaderBag;
+
+        $this->cacheLifetimeEnhancer = new CacheLifetimeEnhancer(
+            $this->cacheLifetimeResolver->reveal(),
+            $this->maxAge,
+            $this->sharedMaxAge
+        );
+    }
+
+    public function provideCacheLifeTime()
+    {
+        return [
+            [50],
+            [500],
+            [0],
+        ];
+    }
+
+    /**
+     * @param $cacheLifetime
+     *
+     * @dataProvider provideCacheLifeTime
+     */
+    public function testEnhance(int $cacheLifetime)
+    {
+        $this->page->getCacheLifeTime()->willReturn(
+            ['type' => CacheLifetimeResolverInterface::TYPE_SECONDS, 'value' => $cacheLifetime]
+        );
+
+        if ($cacheLifetime > 0) {
+            $this->responseHeaderBag->set(AbstractHttpCache::HEADER_REVERSE_PROXY_TTL, $cacheLifetime)->shouldBeCalled();
+            $this->response->setPublic()->shouldBeCalled();
+            $this->response->setMaxAge($this->maxAge)->shouldBeCalled();
+            $this->response->setSharedMaxAge($this->sharedMaxAge)->shouldBeCalled();
+        } else {
+            $this->responseHeaderBag->set(Argument::cetera())->shouldNotBeCalled();
+            $this->response->setPublic()->shouldNotBeCalled();
+            $this->response->setMaxAge(Argument::any())->shouldNotBeCalled();
+            $this->response->setSharedMaxAge(Argument::any())->shouldNotBeCalled();
+        }
+
+        $this->cacheLifetimeResolver->resolve(Argument::cetera())->willReturn($cacheLifetime);
+        $this->cacheLifetimeEnhancer->enhance($this->response->reveal(), $this->page->reveal());
+    }
+
+    public function testEnhanceSnippet()
+    {
+        $this->cacheLifetimeEnhancer->enhance($this->response->reveal(), $this->snippet->reveal());
```

Comment:
Shouldn't you test if something has been called here? Doesn't look like this test does assert anything...